libraries/AP_HAL/tests/test_prescaler.cpp

```diff
@@ -72,4 +110,51 @@ INSTANTIATE_TEST_CASE_P(
                  50000000   // CUAV_GPS
         ));
 
-AP_GTEST_MAIN()
\ No newline at end of file
+AP_GTEST_MAIN()
+
+uint32_t PrescalerParameterizedTestFixture::test_index = 0;
+
+PrescalerParameterizedTestFixture::TestResult PrescalerParameterizedTestFixture::test_results[] = {
+    { 200000000, 1200000, 1197604, 166, DSHOT },
+    { 216000000, 1200000, 1200000, 179, DSHOT },
+    { 108000000, 1200000, 1200000, 89, DSHOT },
+    { 84000000, 1200000, 1200000, 69, DSHOT },
+    { 168000000, 1200000, 1200000, 139, DSHOT },
+    { 50000000, 1200000, 1190476, 41, DSHOT },
+    { 200000000, 2400000, 2409638, 82, DSHOT },
+    { 216000000, 2400000, 2400000, 89, DSHOT },
+    { 108000000, 2400000, 2400000, 44, DSHOT },
+    { 84000000, 2400000, 2400000, 34, DSHOT },
+    { 168000000, 2400000, 2400000, 69, DSHOT },
+    { 50000000, 2400000, 2380952, 20, DSHOT },
+    { 200000000, 4800000, 4761904, 41, DSHOT },
+    { 216000000, 4800000, 4800000, 44, DSHOT },
+    { 108000000, 4800000, 4695652, 22, DSHOT },
+    { 84000000, 4800000, 4666666, 17, DSHOT },
+    { 168000000, 4800000, 4800000, 34, DSHOT },
+    { 50000000, 4800000, 5000000, 9, DSHOT },
+    { 200000000, 9600000, 9523809, 20, DSHOT },
+    { 216000000, 9600000, 9391304, 22, DSHOT },
+    { 108000000, 9600000, 9818181, 10, DSHOT },
+    { 84000000, 9600000, 9333333, 8, DSHOT },
+    { 168000000, 9600000, 9333333, 17, DSHOT },
+    { 50000000, 9600000, 10000000, 4, DSHOT },
+    { 200000000, 192000, 191938, 1041, NONE },
+    { 216000000, 192000, 192000, 1124, NONE },
+    { 108000000, 192000, 192170, 561, NONE },
+    { 84000000, 192000, 192219, 436, NONE },
+    { 168000000, 192000, 192000, 874, NONE },
+    { 50000000, 192000, 192307, 259, NONE },
+    { 200000000, 6400000, 6451612, 30, NEOPIXEL },
+    { 216000000, 6400000, 6352941, 33, NEOPIXEL },
+    { 108000000, 6400000, 6352941, 16, NEOPIXEL },
+    { 84000000, 6400000, 6461538, 12, NEOPIXEL },
+    { 168000000, 6400000, 6461538, 25, NEOPIXEL },
+    { 50000000, 6400000, 6250000, 7, NEOPIXEL },
+    { 200000000, 12000000, 11764705, 16, NONE },
```

Comment:
can we deliberately break one of these values and push up and check CI does fail, I'm just not sure it does

Reply:
Done and it does